retro_data_structures/formats/script_layer.py

```diff
@@ -210,3 +213,27 @@ def is_modified(self) -> bool:
 
     def mark_modified(self):
         self._modified = True
+    
+    def build_mlvl_dependencies(self, asset_manager: AssetManager) -> typing.Iterator[Dependency]:
+        logging.debug(f"        Layer: {self.name}")
+        
+        deps: list[Dependency] = []
+        for instance in self.instances:
+            deps.extend(instance.mlvl_dependencies_for(asset_manager))
+        
+        unique_deps: set[Dependency] = set()
+        for dep in deps:
+            if dep in unique_deps:
+                continue
+            # specifically keep the order of the *first* appearance of the dependency
```

Comment:
does the order matter? because if so, `set` does not guarantee insertion order. That's only `dict`.

Reply:
the order of `unique_deps` doesn't matter. the order in which the dependencies in this function are yielded is what matters